llarp/link/link_manager.cpp

```diff
@@ -0,0 +1,353 @@
+#include <link/link_manager.hpp>
+
+#include <algorithm>
+
+namespace llarp
+{
+
+  LinkLayer_ptr
+  LinkManager::GetCompatibleLink(const RouterContact &rc) const
+  {
+    {
+      util::Lock l(&_mutex);
+      if (stopping) return nullptr;
```

Comment:
you don't need to lock to check an atomic

Reply:
changed.